xdsl/dialects/stencil.py

```diff
@@ -432,6 +479,29 @@ def get(temp: SSAValue | Operation, offset: Sequence[int]):
             result_types=[temp_type.element_type],
         )
 
+    def verify_(self) -> None:
+        apply = self.parent_op()
+        # As promised by HasParent(ApplyOp)
+        assert isinstance(apply, ApplyOp)
+
+        # TODO This should be handled by infra, having a way to verify things on ApplyOp
```

Comment:
Could you file an issue to discuss here? What are the MLIR semantics?

Reply:
The original (not an upstreamed dialect) semantics are a bit different in terms of IR elements, I tweaked a design in the dialect; It was initially expecting an attribute, where now I carry this information in the type.

But what I conceptually want is say "this a variadic result with at least one result". And I want the child operations to be able to assume there's at least one result, so assume some things are verified before verifying the childrens.

MLIR expose both `verify_invariants` and `verify_region` for this nuance precisely in place of our single `verify_` (which is children-first)

Reply:
#1112, not sure about the wording, I rarely work with issues! :upside_down_face: Feel free to rephrase if useful

Reply:
Thank you! The wording is perfect. I really like issues to track todos in code, with a link in the comment, so that when we get to it we can see the history and the discussion. If you think that this particular bit of code should be changed once we have the added methods, could you please file another issue and reference the first one? That way we track everything.

Reply:
I'm not sure what the other issue should be? :thinking: 
I like the idea of adding a link to the first one in the `# TODO` though!

Reply:
It depends on how we split the work when we tackle that issue. It's possible that someone adds the functionality described in your issue, and never updates this line in the code.